operator/iamkeycontroller/create.go

```diff
@@ -54,45 +63,121 @@ func (p *IAMKeyPipeline) Create(ctx context.Context, mg resource.Managed) (manag
 
 // createIAMKey creates a new IAMKey in the project associated with the API Key and Secret.
 func (p *IAMKeyPipeline) createIAMKey(ctx *pipelineContext) error {
+	iamRoleID := ""
+	iamRoleNameFound := false
+
 	iamKey := ctx.iamKey
 	log := controllerruntime.LoggerFrom(ctx)
+	log.Info("starting creation")
+
+	// send request
+	resp, err := ExecuteRequest(ctx, "GET", ctx.iamKey.Spec.ForProvider.Zone, "/v2/iam-role", p.apiKey, p.apiSecret, nil)
+	if err != nil {
+		return err
+	}
+
+	iamRoles := IamRolesList{}
+	// unmarshall response to iamRoles
+
+	err = json.NewDecoder(resp.Body).Decode(&iamRoles)
+	if err != nil {
+		log.Error(err, "Cannot unmarshall response to iamRoles")
+		return err
+	}
 
-	var keyResources []exoscalesdk.IAMAccessKeyResource
-	for _, bucket := range iamKey.Spec.ForProvider.Services.SOS.Buckets {
-		keyResource := exoscalesdk.IAMAccessKeyResource{
-			Domain:       SOSResourceDomain,
-			ResourceName: bucket,
-			ResourceType: BucketResourceType,
+	for _, val := range iamRoles.IamRoles {
```

Comment:
because we do something realy non-crossplane here, I should rather create iamrolecontroleer and create logic for CRUD of iam-role, then add it to composition, so there would be normal Create() and observer() process, since I create 2 things in one call, they depend on each other, by this I tried to ensure that role always exists which was bad approach, it should be fixed now by retrying of IAM-KEY creation

Reply:
If I create separate role, then detection of legacy key and new key would be terribly hard, with current implementation, my goal is to ensure observability and deletion of old keys and CREATE of new types of key, so with time we will simply  replace old keys with newer ones and we can do the refactor